tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordSessionTests.cs

```diff
@@ -127,6 +127,39 @@ public void BodyNormalizationWorksWhenMatching(string body, string contentType)
             Assert.NotNull(deserializedSession.Lookup(recordEntry, matcher, new[] { new RecordedTestSanitizer() }));
         }
 
+        [Theory]
+        [InlineData("<body>data</body>", "text/xml")]
+        [InlineData("{\"json\":\"value\"}", "application/json")]
+        [InlineData("{\"json\":\"value\"}", "application/json+param=val")]
+        [InlineData("missing", null)]
+        public void BodyNormalizationRespectsContentType(string body, string contentType)
+        {
+            byte[] bodyBytes = Encoding.UTF8.GetBytes(body);
+
+            var session = new RecordSession();
+
+            RecordEntry recordEntry = new RecordEntry();
+            if (contentType != null)
+            {
+                recordEntry.Request.Headers.Add("Content-Type", new[] { contentType });
+            }
+
+            recordEntry.Request.Body = bodyBytes;
+            recordEntry.RequestUri = "http://localhost/";
+            recordEntry.RequestMethod = RequestMethod.Delete;
+            RecordEntry.NormalizeJsonBody(recordEntry.Request);
+
+            // Intentionally not using Assert.Equals/NotEquals to check for reference equality
```

Comment:
Assert.AreSame

Reply:
Ah I was looking for the XUnit equivalent, but didn't see it. It's Assert.Same.